spec/models/vm/retirement_management_spec.rb

```diff
@@ -157,4 +157,24 @@
 
     vm.raise_audit_event(event_name, message)
   end
+
+  it "reset retirement state in future" do
+    @vm.update_attributes(:retirement_state => 'retiring')
+    options = { :date => Date.today + 1}
+    @vm.retire(options)
```

Comment:
@mkanoor Is there a reason you are not just calling the `retires_on=` method with the date?

If you do end up keeping this there is no reason to setup the `options` hash, just use: `@vm.retire(:date => Date.today + 1)`